docs/Building.md

```diff
@@ -1,28 +1,173 @@
 
-## Building `calicoctl` CLI Tool - mainline
+### Building `calicoctl` CLI Tool
+
+Our mainline makefile target uses a Debian-based container to build the `calicoctl` tool.  If this is not appropriate
+for your Linux distribution, you can build the tool in a python virtualenv on your distro.  Follow the appropriate
+instructions below.
+
+#### Mainline build (Debian-based container build)
+
+From the root directory of the checked out repository, run
 
     sudo make binary
 
 This builds `calicoctl` command line tool in the `dist/` directory.
 
-## Building `calicoctl` CLI Tool - virtualenv
+#### Virtual environment build
+
+Setup a virtualenv using the instructions detailed [here](#setting-up-a-virtualenv).
+
+Build `calicoctl` with `pyinstaller`.
+
+    pyinstaller calico_containers/calicoctl.py -a -F -s --clean
+
+This builds `calicoctl` command line tool in the `dist/` directory.
+
+
+## The Calico Docker test suites and test environment
+
+There are two test suites used to validate Calico Docker function - unit tests and system tests.  In addition to the
+tests, there make targets to spin up an instance of etcd and consul - useful for test deployments.
+
+If you are developing Calico Docker code that you would like to contribute upstream, please ensure at a minimum that
+the unit tests and system tests successfully run, and, preferably, ensure new functionality is unit tested and system
+tested where necessary.  We aim to keep improving our total code and path coverage with each check-in.
+
+### Spinning up an instance of etcd and consul
+
+The supplied Makefile includes targets for spinning up a single instance of and etcd server and a consul server.  This
+is useful if you want to do any local testing.
+
+The system test make target automatically spins up an instance of etcd and consul which are used in the tests.
+
+#### To start an single etcd server
+
+From the root directory of the checked out repository, run
+
+    make run-etcd
+
+This starts an etcd server bound to a local IP address (automatically determined).  If you have multiple local IP
+addresses and wish to explicitly select an IP address to bind to, set the environment variable LOCAL_IP_ENV with the
+correct IP address.  For example:
+
+    LOCAL_IP_ENV=192.168.0.23 make run-etcd
+
+#### To start an single consul server
+
+From the root directory of the checked out repository, run
+
+    make run-consul
+
+This starts an consul server bound to a local IP address (automatically determined).  If you have multiple local IP
+addresses and wish to explicitly select an IP address to bind to, set the environment variable LOCAL_IP_ENV with the
+correct IP address.  For example:
+
+    LOCAL_IP_ENV=192.168.0.23 make run-etcd
+
+### Running the unit tests
+
+The Calico Docker unit tests provide code and path coverage of the main pycalico modules, a number of the integration
```

Comment:
Nope - some of these tests are moved to libcalico so I think this is wrong. And we'll be moving the other stuff soon so best to just keep this a bit less specific.